moshi/src/test/java/com/squareup/moshi/MoshiTest.java

```diff
@@ -995,6 +995,39 @@
     }
   }
 
+  @Test public void deferredAdapterRecursesInfinitely() throws Exception {
```

Comment:
Deferring StackOverflowErrors isn’t really what we’re going for with this feature.

Instead imagine you’re building an adapter for a binary tree:
```
class TreeNode {
  String value;
  TreeNode left;
  TreeNode right;
}
```
The standard reflective adapter for this is going to recursively look itself up, and it’s going to mostly just work.